### PR TITLE
Make gc more friendly

### PR DESCRIPTION
**Pull request by d-k-c**
This PR intends to give more explicit error messages to people using gc. The goal of the last commit is to allow to use the tool without the need to create a config file.

### CODE REVIEW

File: gc.go [hunk 1]
```diff
@@ -33,7 +33,87 @@ type Config struct {
...
 
+func setupConfig() *Config {
+	var config Config
+
+	configFiles := []string{"${XDG_CONFIG_HOME}/gc/config",
+		"${HOME)/.config/gcrc",
+		"${HOME}/.gcrc"}
+
+	var path string
+	for _, path = range configFiles {
```
Comment (phacops) [inline gc.go]: You don't need to declare `path` above, just `for _, path := range configFiles {`

File: gc.go [hunk 2]
```diff
@@ -33,7 +33,87 @@ type Config struct {
...
+
+			err = json.NewDecoder(configFile).Decode(&config)
+
+			if err != nil {
+				panic(err)
+			}
+
+			break
+		} else {
+			path = ""
```
Comment (phacops) [inline gc.go]: `path` is already initialized and empty.

File: gc.go [hunk 3]
```diff
@@ -33,7 +33,87 @@ type Config struct {
...
+				panic(err)
+			}
+			config.WatchDir = strings.TrimSpace(watchdir)
+		}
+	}
+
+	if config.GarminConnectUsername == "" ||
+		config.GarminConnectPassword == "" ||
+		config.WatchDir == "" {
+		panic("Option(s) missing. Aborting")
```
Comment (phacops) [inline gc.go]: Do we want to `panic` or exit cleanly with a nice error message with `os.Exit(1)`?

File: gc.go (hunk 2 above)
Comment (d-k-c) [reply gc.go]: This is used to bail out early if gc is run with --no-interactive and no config file is present, but I agree this is weird. I'll remove that.

File: gc.go (hunk 3 above)
Comment (d-k-c) [reply gc.go]: Good idea.

**Comment by phacops** [comment]
Thank you!

**Pull request merged**